chapter10/spg/exercise1.pl

```diff
@@ -0,0 +1,27 @@
+#! /usr/bin/perl
+
+use strict;
+use warnings;
+
+# Lazy style
+#chdir;  # the default is our home directory
+#my @sorted = sort { -s $a <=> -s $b } glob '*';
+
+my @files = glob '*';
+
+print "Files in dir:", map { " $_" } @files, "\n";
+
+# Power of the Schwartz
+my @sorted =
+  map { $_->[0] }
+  sort { $a->[1] <=> $b->[1] }
+  map [ $_, get_size( $_ ) ],
+  @files;
+
+print "Files sorted by size:", map { " $_" } @sorted;
```

Comment:
Perl adds spaces if you interpolate a list so you don't need to map (or join), i.e.:
```perl
print "Files: @sorted\n";
```